include/seqan3/contrib/parallel/buffer_queue.hpp

```diff
@@ -191,46 +228,88 @@ class buffer_queue
     bool is_empty() const noexcept
     {
         std::unique_lock write_lock(mutex);
-        return headPos == tailPos;
+        return pop_front_position == push_back_position;
     }
 
     bool is_full() const noexcept
     {
         std::unique_lock write_lock(mutex);
-        return size_impl() == data.max_size();
+        return !has_free_slots(pop_front_position, push_back_position);
```

Comment:
The private member function `has_free_slots` is a simple inequality. Wouldn't it be easier to insert the inequality straight here (and avoid the function call and copying the two variables)?

Reply:
Well, I think reading `push_back_position == pop_front_position + ring_buffer_capacity` is really hard to understand for others. It requires to understand the complete logic of the implementation of the ring buffer which is really hard, especially when trying to reason about certain semantics of the code. So this check is refactored on purpose to make it more understandable what happens here. Also there is now a documentation behind the rational of this implementation.